src/clients/KubectlClient.ts

```diff
@@ -100,62 +100,48 @@ export class KubectlClient implements IClient {
         }
     }
 
-    public async getContainersList(podName: string): Promise<string[]> {
-        if (!podName) {
-            this._logger.error(TelemetryEvent.KubectlClient_GetPodNameError, new Error(`Pod name is null`));
+    public async getContainersList(podNameList: string[], namespace: string): Promise<string[]> {
+        if (!podNameList || podNameList.length == 0) {
+            this._logger.error(TelemetryEvent.KubectlClient_GetPodNameError, new Error(`Pod name List is null`));
             return null;
         }
         try {
             // adding these decent amount of known sidecars to filter out the sidecars from the list
-            const knownSideCars: string[] = ['linkerd-proxy', 'linkerd-init', 'istio-proxy', 'darpd', 'jaeger-agent', 'nginx-proxy'];
-            const kubeconfigPath: string = await this._accountContextManager.getKubeconfigPathAsync();
-            const args: string[] = [`get`, `pod`, podName, `-o`, `jsonpath="{.spec['containers','initContainers'][*].name}"`];
-            const kubectlOutput: string = await this.runKubectlCommandAsync(args, kubeconfigPath);
-            if (kubectlOutput) {
-                // replace the double quotes and split the string by space ex: "linkerd-proxy stats-api linkerd-init" to [linkerd-proxy, stats-api, linkerd-init]
-                // and filter out the known sidecars from the list ex: [stats-api]
-                return kubectlOutput.replace('"', '').replace('"', '').split(' ')
-                    .filter(s => !knownSideCars.find(knownSideCar => knownSideCar == s));
-            } else {
-                this._logger.error(TelemetryEvent.KubectlClient_GetContainerListError, new Error(`container list is null`));
-                return null;
-            }
+            //const knownSideCars: string[] = ['linkerd-proxy', 'linkerd-init', 'istio-proxy', 'darpd', 'jaeger-agent', 'nginx-proxy'];
+            let containerList: string[] = [];
+            const k8sClient = await this._accountContextManager.getK8sClient();
+            const response = await k8sClient.listNamespacedPod(namespace);
+            const pods = response.body.items;
+            podNameList.forEach(podName => {
+                const arr = pods
+                .find(pod => pod.metadata.name === podName)
```

Comment:
Note: this is different from the `kubectl` command because it only looks at `containers` and not `initContainers`. However, I think this is more correct because the [Bridge code](https://github.com/Azure/Bridge-To-Kubernetes/blob/65a0527df3ad85525668c05e8737de71247087ab/src/library/Utilities/RemoteContainerConnectionDetailsResolver.cs#L111) also only considers `containers`, so letting the user pick an `initContainer` would fail.